lib/srv/db/postgres/connector.go

```diff
@@ -0,0 +1,166 @@
+// Teleport
+// Copyright (C) 2024 Gravitational, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package postgres
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/gravitational/trace"
+	"github.com/jackc/pgconn"
+	"github.com/jackc/pgx/v4"
+	"github.com/sirupsen/logrus"
+
+	"github.com/gravitational/teleport/api/types"
+	libcloud "github.com/gravitational/teleport/lib/cloud"
+	"github.com/gravitational/teleport/lib/srv/db/cloud"
+	"github.com/gravitational/teleport/lib/srv/db/common"
+	discoverycommon "github.com/gravitational/teleport/lib/srv/discovery/common"
+)
+
+type connector struct {
+	auth         common.Auth
+	cloudClients libcloud.Clients
+	log          logrus.FieldLogger
+
+	certExpiry    time.Time
+	database      types.Database
+	databaseUser  string
+	databaseName  string
+	startupParams map[string]string
+}
+
+func (c *connector) copy() *connector {
+	return &connector{
+		auth:          c.auth,
+		cloudClients:  c.cloudClients,
+		certExpiry:    c.certExpiry,
+		database:      c.database,
+		databaseUser:  c.databaseUser,
+		databaseName:  c.databaseName,
+		startupParams: c.startupParams,
+	}
+}
```

Comment:
why not copy like this:
https://github.com/gravitational/teleport/blob/0f4eadd8d0a34eb6119e58bc2d9cb3bfc3e87b4b/lib/srv/db/common/session.go#L87-L88
Are we explicitly skipping `log`?

Reply:
Ah, yes that is much better. Harder to forget about fields in this way 😅